core/remote/src/main/kotlin/team/noweekend/core/remote/di/NetworkModule.kt

```diff
@@ -0,0 +1,89 @@
+package team.noweekend.core.remote.di
+
+import dagger.Module
+import dagger.Provides
+import dagger.hilt.InstallIn
+import dagger.hilt.components.SingletonComponent
+import io.ktor.client.HttpClient
+import io.ktor.client.engine.android.Android
+import io.ktor.client.plugins.DefaultRequest
+import io.ktor.client.plugins.HttpTimeout
+import io.ktor.client.plugins.auth.Auth
+import io.ktor.client.plugins.contentnegotiation.ContentNegotiation
+import io.ktor.client.plugins.logging.ANDROID
+import io.ktor.client.plugins.logging.LogLevel
+import io.ktor.client.plugins.logging.Logger
+import io.ktor.client.plugins.logging.Logging
+import io.ktor.client.plugins.resources.Resources
+import io.ktor.http.ContentType
+import io.ktor.http.URLProtocol
+import io.ktor.http.contentType
+import io.ktor.serialization.kotlinx.json.json
+import kotlinx.serialization.json.Json
+import team.noweekend.core.remote.BuildConfig
+import team.noweekend.core.remote.provider.AuthenticationProvider
+import team.noweekend.core.remote.qulifier.BaseUrl
+import team.noweekend.core.remote.qulifier.BasicClient
+import javax.inject.Singleton
+
+@Module
+@InstallIn(SingletonComponent::class)
+internal class NetworkModule {
+    
+    @Provides
+    @Singleton
+    @BaseUrl
```

Comment:
p2: 요거 Qualifier는 어떤 목적인가욤?

Reply:
일단은 하드 코딩 방지 목적이 젤 컸구 향후 url이 여러개가 있을 경우, 디버그,qa,릴리즈 모드에 따라서 url이 달라질 경우를 대비해서 확장서 있게 설정해 놓은거긴행

우리 플젝 같은 경우에는 서버가 여러대 있을 경우는 없으니깐 딱히 안써도 되는 거는 맞지만은? 어노테이션으로 사용하고 관리할 수 있다는 점도 장점 같아서! 다른 더 나은 방식이 있는걸까?

Reply:
이렇게 작성될 수도 있을 것 같은 느낌? 하드코딩 방지가 Flavor 하드코딩을 말하는거면 저거는 따로 enum으로 정의해서 수정해볼겝
```suggestion
    @Provides
    @Singleton
    @BaseUrl
    fun provideServerHost(): String {
        return when(BuildConfig.FLAVOR) {
            "dev", "qa" -> "AAAA.com"
            "prod" -> "BBBB.com"
             else -> throw IllegalArgumentException()
        }
    }
```

Reply:
그러고보니 BuildConfig.Flavor는 develop 브랜치 헤드로 지금 브랜치 맞춰놓으면 사용 가능!

Reply:
@JaesungLeee ㅇㅋㅇㅋ 그러면 일단은 Qulifier 지우고 나중에 enum으로 수정하는 방식으로 갈까?

Reply:
조아 enum으로 교체하는 코드는 수정해둘겝!

Reply:
@JaesungLeee https://github.com/mash-up-kr/NoWeekend-Android/pull/23/commits/ef006a9568aef2166d5c4ba5cb124af41ff7d88c 반영 완~